pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go

```diff
@@ -159,7 +159,7 @@ func (w *Watcher) traversePluginDir(dir string) error {
 func (w *Watcher) handleCreateEvent(event fsnotify.Event) error {
 	klog.V(6).Infof("Handling create event: %v", event)
 
```

Comment:
this breaks plugins that register in the following way:
```
plugin sets up a dir on disk somewhere:
/myplugin/mysocket.sock

then symlinks its dir into the kubelet's plugin dir:
/var/plugindir/myplugin -> /myplugin
```

`event.Name` would be `/var/plugindir/myplugin`

previously:
* `os.Stat` would follow, and get info for /var/plugindir/myplugin
* fi.IsDir() would be true
* we would traverse and find the plugin socket on line 185

with this change:
* `os.Lstat` does not follow
* `fi.IsDir()` is false
* `IsUnixDomainSocket` is also false
* we ignore and return "klog.V(5).Infof("Ignoring non socket file %s", fi.Name())"

If os.Stat has known issues with sockets on windows, I think this would be a better workaround:

```go
fi, err := os.Stat(event.Name)
if err != nil && os.GOOS == "windows" {
  fi, err = os.Lstat(event.Name)
}
```

Reply:
created a PR to address https://github.com/kubernetes/kubernetes/pull/99723
I missed the traverse part. Thank you for pointing it out!

Reply:
I tested this logic on Windows and found out in the case path is a symlink points to a dir, os.Stat() will not work for all Windows versions tested including LSTC and SAC

The error is "Access is Denied".
So this use case will not work for Windows before or after this change.